binder/mapping_test.go

```
@@ -177,3 +178,76 @@
+			t.Fatalf("unexpected error: %v", err)
+		}
+		if len(data["files"]) != 2 {
+			t.Fatalf("expected 2 files, got %d", len(data["files"]))
+		}
+	})
+
+	t.Run("type casting error", func(t *testing.T) {
+		out := struct{}{}
+		data := map[string][]int{} // Incorrect type to force a casting error
+		err := formatBindData(out, data, "key", "value", false, false)
+		require.Equal(t, err.Error(), "unsupported value type: string")
```

golangci-lint (1.62.2)

expected-actual: need to reverse actual and expected values

(testifylint)